packages/utils/src/lib/formatting.ts

```diff
@@ -71,12 +71,40 @@ export function formatDate(date: Date): string {
     .replace(/\u202F/g, ' '); // see https://github.com/nodejs/node/issues/45171
 }
 
-export function truncateText(text: string, maxChars: number): string {
+export function truncateText(
+  text: string,
+  options:
+    | number
+    | {
+        maxChars: number;
+        position?: 'start' | 'middle' | 'end';
+        ellipsis?: string;
+      },
+): string {
+  const {
+    maxChars,
+    position = 'end',
+    ellipsis = '...',
+  } = typeof options === 'number' ? { maxChars: options } : options;
   if (text.length <= maxChars) {
     return text;
   }
-  const ellipsis = '...';
-  return text.slice(0, maxChars - ellipsis.length) + ellipsis;
+
+  const maxLength = maxChars - ellipsis.length;
+  switch (position) {
+    case 'start':
+      return ellipsis + text.slice(-maxLength).trim();
+    case 'middle':
+      const halfMaxChars = Math.floor(maxLength / 2);
+      return (
+        text.slice(0, halfMaxChars).trim() +
+        ellipsis +
+        text.slice(-halfMaxChars).trim()
+      );
+    // case 'end':
+    default:
```

Comment:
Checking for `'end'` explicitly is more future-proof:

```suggestion
    case 'end':
```